src/components/generate/GeneratePlaylistStepper.tsx

```diff
@@ -6,81 +9,141 @@ import { Stepper, StepList, Step, StepContent } from "@components/ui/Stepper"
 import useTranslation from "next-translate/useTranslation"
 import Link from "next/link"
 import { useState } from "react"
+import { Form } from "@components/ui/Form"
 
 const STEPS_COUNT = 3
 
+const formSchema = z
+  .object({
+    playlistType: z.enum(["new", "existing"]),
+    name: z.string().optional(),
+    playlistSelected: z.string().optional(),
+    isPrivate: z.boolean(),
+    bands: z.array(z.number().min(1)).nonempty("At least one band is required"),
+  })
+  .superRefine((data, ctx) => {
+    if (data.playlistType === "new" && !data.name) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "Name is required for NEW role",
```

Comment:
What do we mean by role, here? Should we also use lowercaps for the playlist type? We can quote the string, instead.